app/main.py

```diff
@@ -1,3 +1,27 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    memory_cache = {}
+
+    def inner(*args, **kwargs) -> list and int:
+        nonlocal memory_cache
+        immute = (dict, set, list)
+        for argumets in args:
+            if type(argumets) in immute:
+                return
```

Comment:
U don't need to check the args type 
![image](https://user-images.githubusercontent.com/103201010/204539467-13905180-752f-4656-9a03-afa3e268a7e1.png)